cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -505,9 +502,10 @@ public void upgradeApplication(HttpRequest request, HttpResponder responder,
                                  @PathParam("app-id") String appName,
                                  @QueryParam("artifactScope") Set<String> artifactScopes,
                                  @QueryParam("allowSnapshot") boolean allowSnapshot) throws Exception {
-    ApplicationId appId = validateApplicationId(validateNamespace(namespaceId), appName);
+    validateApplicationId(namespaceId, appName);
     Set<ArtifactScope> allowedArtifactScopes = getArtifactScopes(artifactScopes);
-    applicationLifecycleService.upgradeApplication(appId, allowedArtifactScopes, allowSnapshot);
+    ApplicationId appId = new ApplicationId(namespaceId, appName);
+    appId = applicationLifecycleService.upgradeApplication(appId, allowedArtifactScopes, allowSnapshot);
```

Comment:
I don't think this would work.
First thing `upgradeApplication` does is calling `store.getApplicationMetadata(appId);`. I believe it would fail for `-SNASHOT`. And if not - we should pass the namespace/appName

Reply:
We will have a follow up discussion on /upgrade and /update https://cdap.atlassian.net/browse/CDAP-19782